contracts/utils/Roles.sol

```diff
@@ -0,0 +1,359 @@
+
+    /**
+     * @notice Modifier that throws an error if called by an account without all the roles.
+     */
+    modifier withAllTheRoles(uint8[] calldata roles) {
+        if (!hasAllTheRoles(msg.sender, roles)) {
+            revert MissingAllRoles(msg.sender, roles);
+        }
+
+        _;
+    }
+
+    /**
+     * @notice Modifier that throws an error if called by an account without any of the roles.
+     */
+    modifier withAnyOfRoles(uint8[] calldata roles) {
+        if (!hasAnyOfRoles(msg.sender, roles)) {
+            revert MissingAnyOfRoles(msg.sender, roles);
+        }
+
+        _;
+    }
+
+    /**
+     * @notice Checks if an account has a role.
+     * @param account The address to check
+     * @param role The role to check
+     * @return True if the account has the role, false otherwise
+     */
+    function hasRole(address account, uint8 role) public view returns (bool) {
+        return _hasRole(_getRoles(account), role);
+    }
+
+    /**
+     * @notice Checks if an account has all the roles.
+     * @param account The address to check
+     * @param roles The roles to check
+     * @return True if the account has all the roles, false otherwise
+     */
+    function hasAllTheRoles(address account, uint8[] calldata roles) public view returns (bool) {
+        return _hasAllTheRoles(_getRoles(account), roles);
+    }
+
+    /**
+     * @notice Checks if an account has any of the roles.
+     * @param account The address to check
+     * @param roles The roles to check
+     * @return True if the account has any of the roles, false otherwise
+     */
+    function hasAnyOfRoles(address account, uint8[] calldata roles) public view returns (bool) {
+        return _hasAnyOfRoles(_getRoles(account), roles);
+    }
+
+    /**
+     * @notice Returns the roles of an account.
+     * @param account The address to get the roles for
+     * @return accountRoles The roles of the account in uint256 format
+     */
+    function getAccountRoles(address account) public view returns (uint256 accountRoles) {
+        accountRoles = _getRoles(account);
+    }
+
+    /**
+     * @notice Returns the pending role of the contract.
+     * @param fromAccount The address of the current role
+     * @param toAccount The address of the pending role
+     * @return proposedRoles_ The pending role of the contract in uint256 format
+     */
+    function proposedRoles(address fromAccount, address toAccount) public view returns (uint256 proposedRoles_) {
+        proposedRoles_ = _getProposedRoles(fromAccount, toAccount);
+    }
+
+    /**
+     * @notice Transfers roles of the contract to a new account.
+     * @dev Can only be called by the account with all the roles.
+     * @dev Emits RolesRemoved and RolesAdded events.
+     * @param toAccount The address to transfer role to
+     * @param roles The roles to transfer
+     */
+    function transferRoles(address toAccount, uint8[] calldata roles) external virtual withAllTheRoles(roles) {
```

Comment:
Do we want to add the same sanity checks as `proposeRoles`?
`if (toAccount == address(0) || toAccount == msg.sender) revert InvalidTransferAccount(toAccount);`